src/csm_testing/tests/sat_functional/nid2xname/test_nid2xname.py

```diff
@@ -30,140 +30,144 @@
 
 class TestNid2Xname(SATTestCase):
 
+    @classmethod
+    def setUpClass(cls):
+        """Set up data used by all test methods."""
+        super().setUpClass()
+        cls.components = cls.get_hsm_components()
+        # Pre-calculate the first node's data for simple tests
+        cls.first_xname = cls.components[0]['Xname']
+        cls.first_nid = cls.components[0]['NID']
+
+    @classmethod
+    def get_hsm_components(cls):
+        """Get component data from HSM."""
+        # Get raw HSM data - no jq processing needed
+        command = "cray hsm state components list --type Node --format json"
+        result = subprocess.run(command, shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        # Process the data in Python instead of jq
+        raw_data = json.loads(result.stdout.decode().strip())
+        components = []
+
+        # Extract just the NID and Xname (ID) fields from each component
+        for component in raw_data.get('Components', []):
+            if 'NID' in component and 'ID' in component:
+                components.append({
+                    'NID': component['NID'],
+                    'Xname': component['ID']
+                })
+
+        # Sort by NID (as integers)
+        components.sort(key=lambda c: int(c['NID']) if c['NID'] is not None else float('inf'))
+
+        return components
+
+    @classmethod
+    def get_consecutive_components(cls, min_count=3):
+        """Return a list of components with at least min_count consecutive NIDs."""
+        nids_int = [int(c['NID']) for c in cls.components]
+        for i in range(len(nids_int) - (min_count - 1)):
+            if all(nids_int[i + j] == nids_int[i] + j for j in range(min_count)):
+                indices = list(range(i, i + min_count))
+                return [cls.components[idx] for idx in indices]
+        raise AssertionError(f"Could not find {min_count} consecutive NIDs in HSM output")
+
     def run_command(self, command):
         """Run a shell command and return the output."""
         result = subprocess.run(command, shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
I don't believe there is any need for `shell=True` here anymore.


```suggestion
        result = subprocess.run(command, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```